### PR TITLE
Allow format.json healing on test setups

### PR DESCRIPTION
**Pull request by krishnasrinivas**
## Description

Currently we disallow healing on root disks. But for dev/test setups (single node XL, all root disks) we can make it an exception. (to make testing of healing easy)

## Regression

No

## How Has This Been Tested?

tested by simulating disk mount using loop back devices.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added unit tests to cover my changes.
- [ ] I have added/updated functional tests in [mint](https://github.com/minio/mint). (If yes, add `mint` PR # here: )
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by minio-ops** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-tls.sh | :heavy_check_mark: |
| mint-compression-xl.sh | :heavy_check_mark: |
| mint-xl.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-compression-fs.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |

**Review by harshavardhana — APPROVED** [review]
LGTM & tested

**Review by harshavardhana — APPROVED** [review]
LGTM & tested

File: cmd/xl-sets.go [hunk 1]
```diff
@@ -1007,6 +1007,36 @@ func (s *xlSets) ReloadFormat(ctx context.Context, dryRun bool) (err error) {
 	return nil
 }
 
+// If it is a single node XL and all disks are root disks, it is most likely a dev setup, else it is a production setup.
+// On a dev setup we allow creation of format.json on root disks to help with testing.
+func isProductionSetup(infos []DiskInfo, errs []error) bool {
+	rootDiskCount := 0
+	for i := range errs {
+		if errs[i] != nil {
+			continue
```
Comment (kannappanr) [inline cmd/xl-sets.go]: If we are checking for `globalIsDistXL` in the top, then we can return true here also, instead of continuing to loop.

Comment (harshavardhana) [reply cmd/xl-sets.go]: This doesn't need to be because for condition `rootDiskCount == len(errs)`

File: cmd/xl-sets.go [hunk 2]
```diff
@@ -1007,6 +1007,36 @@ func (s *xlSets) ReloadFormat(ctx context.Context, dryRun bool) (err error) {
 	return nil
 }
 
+// If it is a single node XL and all disks are root disks, it is most likely a dev setup, else it is a production setup.
+// On a dev setup we allow creation of format.json on root disks to help with testing.
+func isProductionSetup(infos []DiskInfo, errs []error) bool {
+	rootDiskCount := 0
```
Comment (kannappanr) [inline cmd/xl-sets.go:1016]: We can check if `globalIsDistXL` is true and then return true here.

File: cmd/xl-sets.go [hunk 3]
```diff
@@ -1007,6 +1007,36 @@ func (s *xlSets) ReloadFormat(ctx context.Context, dryRun bool) (err error) {
...
+
+func getAllDiskInfos(storageDisks []StorageAPI) ([]DiskInfo, []error) {
+	infos := make([]DiskInfo, len(storageDisks))
+	errs := make([]error, len(storageDisks))
+	var wg sync.WaitGroup
+	for i := range storageDisks {
+		wg.Add(1)
+		go func(i int) {
+			defer wg.Done()
+			infos[i], errs[i] = storageDisks[i].DiskInfo()
```
Comment (vadmeste) [inline cmd/xl-sets.go:1042]: storageDisks[i] is sometimes nil

**Review by vadmeste — CHANGES_REQUESTED** [review]
One needed change

File: cmd/xl-sets.go [hunk 4]
```diff
@@ -1029,15 +1059,18 @@ func (s *xlSets) HealFormat(ctx context.Context, dryRun bool) (res madmin.HealRe
...
-	for i, disk := range storageDisks {
-		info, err := disk.DiskInfo()
-		if err != nil {
-			storageDisks[i] = nil
-		}
-		if info.RootDisk {
-			// We should not heal on root disk. i.e in a situation where the minio-administrator has unmounted a
-			// defective drive we should not heal a path on the root disk.
-			storageDisks[i] = nil
+	infos, errs := getAllDiskInfos(storageDisks)
```
Comment (harshavardhana) [inline cmd/xl-sets.go]: Actually, we should avoid getAllDiskInfos() for distributed setup, i.e such that following code inside `isProductionSetup` block can be simply executed with this detection.

**Review by harshavardhana — CHANGES_REQUESTED** [review]

File: cmd/xl-sets.go (hunk 2 above)
Comment (krishnasrinivas) [reply cmd/xl-sets.go:1016]: this is taken care of now

File: cmd/xl-sets.go (hunk 3 above)
Comment (krishnasrinivas) [reply cmd/xl-sets.go:1042]: this is taken care of now

File: cmd/xl-sets.go (hunk 4 above)
Comment (krishnasrinivas) [reply cmd/xl-sets.go]: Did not get you @harshavardhana

**Comment by minio-ops** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-xl.sh | :heavy_exclamation_mark: [more...](#60946234-1694-4c44-a4dc-c9eabf972d82) |
| mint-large-bucket.sh | :heavy_exclamation_mark: [more...](#df91c1c4-0551-40b2-bd1b-83d3365f56a7) |
| mint-compression-dist-xl.sh | :heavy_exclamation_mark: [more...](#8b7b9bf0-c8dc-47aa-ab35-e43354babbe8) |
| mint-compression-fs.sh | :heavy_exclamation_mark: [more...](#36a00832-9764-4a5e-9961-9ffb9b579b91) |
| mint-worm.sh | :heavy_exclamation_mark: [more...](#45dc6f9e-8e9a-4fef-85bc-d520e11e4b27) |
| mint-fs.sh | :heavy_exclamation_mark: [more...](#26eac349-2e12-4de8-8011-bf9169fbd4b5) |
| mint-dist-xl.sh | :heavy_exclamation_mark: [more...](#501bc3a2-8bb3-41e8-8833-89aef1f47ae7) |
| mint-gateway-nas.sh | :heavy_exclamation_mark: [more...](#05a788d9-1988-4edc-946a-6d81f1540f20) |
<a name="36a00832-9764-4a5e-9961-9ffb9b579b91"></a>
### 7170-6c08358/mint-compression-fs.sh.xtrace:
```
DISABLED TIME BEING
```
<a name="05a788d9-1988-4edc-946a-6d81f1540f20"></a>
### 7170-6c08358/mint-gateway-nas.sh.xtrace:
```
DISABLED TIME BEING
```
<a name="60946234-1694-4c44-a4dc-c9eabf972d82"></a>
### 7170-6c08358/mint-xl.sh.xtrace:
```
DISABLED TIME BEING
```
<a name="8b7b9bf0-c8dc-47aa-ab35-e43354babbe8"></a>
### 7170-6c08358/mint-compression-dist-xl.sh.xtrace:
```
DISABLED TIME BEING
```
<a name="45dc6f9e-8e9a-4fef-85bc-d520e11e4b27"></a>
### 7170-6c08358/mint-worm.sh.xtrace:
```
DISABLED TIME BEING
```
<a name="26eac349-2e12-4de8-8011-bf9169fbd4b5"></a>
### 7170-6c08358/mint-fs.sh.xtrace:
```
DISABLED TIME BEING
```
<a name="501bc3a2-8bb3-41e8-8833-89aef1f47ae7"></a>
### 7170-6c08358/mint-dist-xl.sh.xtrace:
```
DISABLED TIME BEING
```
<a name="df91c1c4-0551-40b2-bd1b-83d3365f56a7"></a>
### 7170-6c08358/mint-large-bucket.sh.xtrace:
```
DISABLED TIME BEING
```

File: cmd/xl-sets.go (hunk 4 above)
Comment (harshavardhana) [reply cmd/xl-sets.go]: We should avoid getAllDisksInfo() because we need to only allow this for non-distributed setups.

Comment (krishnasrinivas) [reply cmd/xl-sets.go]: no, getAllDisksInfo() is needed for distributed setup so that we dont heal root disks.

Comment (harshavardhana) [reply cmd/xl-sets.go]: Yes but it shouldn't be needed in other scenarios. The point I am saying is if we are going to treat DistXL as production setup always. Why do you need DisksInfo()?

**Comment by minio-ops** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-compression-xl.sh | :x: [more...](#f70ee396-d4c8-412d-8815-9e1e6c5382cd) |
| mint-xl.sh | :x: [more...](#357f188e-0d39-43da-b85c-398a7f92af53) |
| mint-large-bucket.sh | :x: [more...](#a8e87a1e-c9c2-4062-82c7-53339091c1d7) |
| mint-compression-dist-xl.sh | :x: [more...](#bb95933a-8b63-47df-b84a-3b3045cc40a2) |
| mint-compression-fs.sh | :x: [more...](#ebe9d06d-c0eb-400c-8434-b23f680a954b) |
| mint-worm.sh | :x: [more...](#f0c27ca7-cfa6-4f32-9669-de758b222d49) |
| mint-fs.sh | :x: [more...](#a91bf008-717e-4b18-b1d5-f0bf856207c9) |
| mint-dist-xl.sh | :x: [more...](#bceda19e-4250-48c3-857c-a2d3c6460a2e) |
| mint-gateway-nas.sh | :x: [more...](#e9094c8c-1890-4c73-a11c-e688524665d9) |
<a name="ebe9d06d-c0eb-400c-8434-b23f680a954b"></a>
### 7170-a728ba2/mint-compression-fs.sh.log:
```
Running with
SERVER_ENDPOINT:      localhost:31355
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 9838dc7303b7:/mint/log /tmp/mint-logs'

(1/13) Running aws-sdk-go tests ... FAILED in 1 seconds
{
  "alert": "",
  "args": {
    "bucketName": "aws-sdk-go-test-rdkwoxz3lyyqzj",
    "expiry": 60000000000,
    "objectName": "presignedTest"
  },
  "duration": 332,
  "error": "RequestError: send request failed\ncaused by: Put http://localhost:31355/aws-sdk-go-test-rdkwoxz3lyyqzj: dial tcp [::1]:31355: connect: connection refused",
  "function": "PresignedPut",
  "message": "AWS SDK Go CreateBucket Failed",
  "name": "aws-sdk-go",
  "status": "FAIL"
}

Executed 0 out of 13 tests successfully.
```
<a name="a91bf008-717e-4b18-b1d5-f0bf856207c9"></a>
### 7170-a728ba2/mint-fs.sh.log:
```
Running with
SERVER_ENDPOINT:      localhost:31095
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp a7f79c98d13a:/mint/log /tmp/mint-logs'

(1/13) Running aws-sdk-go tests ... FAILED in 0 seconds
{
  "alert": "",
  "args": {
    "bucketName": "aws-sdk-go-test-jocxo5x2hcvptr",
    "expiry": 60000000000,
    "objectName": "presignedTest"
  },
  "duration": 279,
  "error": "RequestError: send request failed\ncaused by: Put http://localhost:31095/aws-sdk-go-test-jocxo5x2hcvptr: dial tcp [::1]:31095: connect: connection refused",
  "function": "PresignedPut",
  "message": "AWS SDK Go CreateBucket Failed",
  "name": "aws-sdk-go",
  "status": "FAIL"
}

Executed 0 out of 13 tests successfully.
```
<a name="f0c27ca7-cfa6-4f32-9669-de758b222d49"></a>
### 7170-a728ba2/mint-worm.sh.log:
```
Running with
SERVER_ENDPOINT:      localhost:32736
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            worm
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp d87202310ed6:/mint/log /tmp/mint-logs'

(1/1) Running worm tests ... FAILED in 1 seconds
{
  "alert": "",
  "args": {
    "bucketName": "bucket-zh00bd7qj0sh6dcdo7e1ewubee4d4c3qtbzabb6k7dajq7goe88d0b4",
    "expiry": 60000000000,
    "objectName": "testObject"
  },
  "duration": 347,
  "error": "RequestError: send request failed\ncaused by: Put http://localhost:32736/bucket-zh00bd7qj0sh6dcdo7e1ewubee4d4c3qtbzabb6k7dajq7goe88d0b4: dial tcp [::1]:32736: connect: connection refused",
  "function": "PutAndDelete",
  "message": "WORM_MODE ON  - CreateBucket Failed",
  "name": "test worm mode",
  "status": "FAIL"
}

Executed 0 out of 1 tests successfully.
```
<a name="a8e87a1e-c9c2-4062-82c7-53339091c1d7"></a>
### 7170-a728ba2/mint-large-bucket.sh.log:
```
Running with
SERVER_ENDPOINT:      localhost:31107
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 3e1eab18db94:/mint/log /tmp/mint-logs'

(1/13) Running aws-sdk-go tests ... FAILED in 0 seconds
{
  "alert": "",
  "args": {
    "bucketName": "aws-sdk-go-test-tg0y96d6w3mtow",
    "expiry": 60000000000,
    "objectName": "presignedTest"
  },
  "duration": 308,
  "error": "RequestError: send request failed\ncaused by: Put http://localhost:31107/aws-sdk-go-test-tg0y96d6w3mtow: dial tcp [::1]:31107: connect: connection refused",
  "function": "PresignedPut",
  "message": "AWS SDK Go CreateBucket Failed",
  "name": "aws-sdk-go",
  "status": "FAIL"
}

Executed 0 out of 13 tests successfully.
```
<a name="bceda19e-4250-48c3-857c-a2d3c6460a2e"></a>
### 7170-a728ba2/mint-dist-xl.sh.log:
```
Running with
SERVER_ENDPOINT:      localhost:31684
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp e187b7a845e3:/mint/log /tmp/mint-logs'

(1/13) Running aws-sdk-go tests ... FAILED in 0 seconds
{
  "alert": "",
  "args": {
    "bucketName": "aws-sdk-go-test-hafa694q4bnd4s",
    "expiry": 60000000000,
    "objectName": "presignedTest"
  },
  "duration": 318,
  "error": "RequestError: send request failed\ncaused by: Put http://localhost:31684/aws-sdk-go-test-hafa694q4bnd4s: dial tcp [::1]:31684: connect: connection refused",
  "function": "PresignedPut",
  "message": "AWS SDK Go CreateBucket Failed",
  "name": "aws-sdk-go",
  "status": "FAIL"
}

Executed 0 out of 13 tests successfully.
```
<a name="f70ee396-d4c8-412d-8815-9e1e6c5382cd"></a>
### 7170-a728ba2/mint-compression-xl.sh.log:
```
Running with
SERVER_ENDPOINT:      localhost:31416
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 0dfef2a3ff32:/mint/log /tmp/mint-logs'

(1/13) Running aws-sdk-go tests ... FAILED in 0 seconds
{
  "alert": "",
  "args": {
    "bucketName": "aws-sdk-go-test-zbezmprsmjglon",
    "expiry": 60000000000,
    "objectName": "presignedTest"
  },
  "duration": 232,
  "error": "RequestError: send request failed\ncaused by: Put http://localhost:31416/aws-sdk-go-test-zbezmprsmjglon: dial tcp [::1]:31416: connect: connection refused",
  "function": "PresignedPut",
  "message": "AWS SDK Go CreateBucket Failed",
  "name": "aws-sdk-go",
  "status": "FAIL"
}

Executed 0 out of 13 tests successfully.
```
<a name="357f188e-0d39-43da-b85c-398a7f92af53"></a>
### 7170-a728ba2/mint-xl.sh.log:
```
Running with
SERVER_ENDPOINT:      localhost:32257
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 4186ffe074aa:/mint/log /tmp/mint-logs'

(1/13) Running aws-sdk-go tests ... FAILED in 0 seconds
{
  "alert": "",
  "args": {
    "bucketName": "aws-sdk-go-test-ss43ts1w4yw51h",
    "expiry": 60000000000,
    "objectName": "presignedTest"
  },
  "duration": 311,
  "error": "RequestError: send request failed\ncaused by: Put http://localhost:32257/aws-sdk-go-test-ss43ts1w4yw51h: dial tcp [::1]:32257: connect: connection refused",
  "function": "PresignedPut",
  "message": "AWS SDK Go CreateBucket Failed",
  "name": "aws-sdk-go",
  "status": "FAIL"
}

Executed 0 out of 13 tests successfully.
```
<a name="bb95933a-8b63-47df-b84a-3b3045cc40a2"></a>
### 7170-a728ba2/mint-compression-dist-xl.sh.log:
```
Running with
SERVER_ENDPOINT:      localhost:30805
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp bcc8b2cca822:/mint/log /tmp/mint-logs'

(1/13) Running aws-sdk-go tests ... FAILED in 0 seconds
{
  "alert": "",
  "args": {
    "bucketName": "aws-sdk-go-test-lb9j4vgrvntnys",
    "expiry": 60000000000,
    "objectName": "presignedTest"
  },
  "duration": 296,
  "error": "RequestError: send request failed\ncaused by: Put http://localhost:30805/aws-sdk-go-test-lb9j4vgrvntnys: dial tcp [::1]:30805: connect: connection refused",
  "function": "PresignedPut",
  "message": "AWS SDK Go CreateBucket Failed",
  "name": "aws-sdk-go",
  "status": "FAIL"
}

Executed 0 out of 13 tests successfully.
```
<a name="e9094c8c-1890-4c73-a11c-e688524665d9"></a>
### 7170-a728ba2/mint-gateway-nas.sh.log:
```
Running with
SERVER_ENDPOINT:      localhost:30085
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp e7e730cbe1e2:/mint/log /tmp/mint-logs'

(1/13) Running aws-sdk-go tests ... FAILED in 0 seconds
{
  "alert": "",
  "args": {
    "bucketName": "aws-sdk-go-test-agh5upm9iboznr",
    "expiry": 60000000000,
    "objectName": "presignedTest"
  },
  "duration": 347,
  "error": "RequestError: send request failed\ncaused by: Put http://localhost:30085/aws-sdk-go-test-agh5upm9iboznr: dial tcp [::1]:30085: connect: connection refused",
  "function": "PresignedPut",
  "message": "AWS SDK Go CreateBucket Failed",
  "name": "aws-sdk-go",
  "status": "FAIL"
}

Executed 0 out of 13 tests successfully.
```

File: cmd/xl-sets.go (hunk 4 above)
Comment (krishnasrinivas) [reply cmd/xl-sets.go]: as disucssed in person, this is needed.

**Review by kannappanr — DISMISSED** [review]
LGTM

**Review by harshavardhana — DISMISSED** [review]

**Comment by minio-ops** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-compression-xl.sh | :heavy_check_mark: |
| mint-xl.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-compression-dist-xl.sh | :heavy_check_mark: |
| mint-compression-fs.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :x: [more...](#28eddb4c-ce7a-43e7-b833-40f1b6d1c2ac) |
| mint-gateway-nas.sh | :x: [more...](#d3945606-c620-47ef-b8d8-c4472bfed187) |
<a name="d3945606-c620-47ef-b8d8-c4472bfed187"></a>
### 7170-1583a6d/mint-gateway-nas.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.52:30521
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 6566994957fc:/mint/log /tmp/mint-logs'

(1/13) Running aws-sdk-go tests ... done in 0 seconds
(2/13) Running aws-sdk-java tests ... done in 1 seconds
(3/13) Running aws-sdk-php tests ... done in 41 seconds
(4/13) Running aws-sdk-ruby tests ... FAILED in 2 seconds
{
  "name": "aws-sdk-ruby",
  "function": "presignedPost(bucket_name,file_name,expires_in_sec,max_byte_size)",
  "args": {
    "bucket_name": "aws-sdk-ruby-bucket-51c30dc2f857",
    "file_name": "datafile-1-MB",
    "expires_in_sec": 60,
    "max_byte_size": 3145728
  },
  "duration": 49.78,
  "error": "Expected to be created object does NOT exist",
  "status": "FAIL"
}

Executed 3 out of 13 tests successfully.
```
<a name="28eddb4c-ce7a-43e7-b833-40f1b6d1c2ac"></a>
### 7170-1583a6d/mint-dist-xl.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.52:30452
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 2a9b7db37742:/mint/log /tmp/mint-logs'

(1/13) Running aws-sdk-go tests ... done in 0 seconds
(2/13) Running aws-sdk-java tests ... done in 1 seconds
(3/13) Running aws-sdk-php tests ... done in 45 seconds
(4/13) Running aws-sdk-ruby tests ... done in 4 seconds
(5/13) Running awscli tests ... done in 1 minutes and 57 seconds
(6/13) Running mc tests ... done in 33 seconds
(7/13) Running minio-dotnet tests ... FAILED in 2 minutes and 52 seconds

Unhandled Exception: System.AggregateException: One or more errors occurred. (One or more errors occurred. (Minio API responded with message=Unsuccessful response from server without XML error: An error occurred while sending the request. The server returned an invalid or unrecognized response.) (Minio API responded with message=Unsuccessful response from server without XML error: An error occurred while sending the request. The server returned an invalid or unrecognized response.)) ---> System.AggregateException: One or more errors occurred. (Minio API responded with message=Unsuccessful response from server without XML error: An error occurred while sending the request. The server returned an invalid or unrecognized response.) (Minio API responded with message=Unsuccessful response from server without XML error: An error occurred while sending the request. The server returned an invalid or unrecognized response.) ---> Minio.Exceptions.InternalClientException: Minio API responded with message=Unsuccessful response from server without XML error: An error occurred while sending the request. The server returned an invalid or unrecognized response.
   at Minio.MinioClient.ParseError(IRestResponse response) in /q/.q/sources/minio-dotnet/Minio/MinioClient.cs:line 458
   at Minio.MinioClient.HandleIfErrorResponse(IRestResponse response, IEnumerable`1 handlers, DateTime startTime) in /q/.q/sources/minio-dotnet/Minio/MinioClient.cs:line 513
   at Minio.MinioClient.ExecuteTaskAsync(IEnumerable`1 errorHandlers, IRestRequest request, CancellationToken cancellationToken) in /q/.q/sources/minio-dotnet/Minio/MinioClient.cs:line 359
   at Minio.MinioClient.PutObjectAsync(String bucketName, String objectName, String uploadId, Int32 partNumber, Byte[] data, Dictionary`2 metaData, Dictionary`2 sseHeaders, CancellationToken cancellationToken) in /q/.q/sources/minio-dotnet/Minio/ApiEndpoints/ObjectOperations.cs:line 492
   at Minio.MinioClient.PutObjectAsync(String bucketName, String objectName, Stream data, Int64 size, String contentType, Dictionary`2 metaData, ServerSideEncryption sse, CancellationToken cancellationToken) in /q/.q/sources/minio-dotnet/Minio/ApiEndpoints/ObjectOperations.cs:line 254
   at Minio.Functional.Tests.FunctionalTest.PutObject_Task(MinioClient minio, String bucketName, String objectName, String fileName, String contentType, Int64 size, Dictionary`2 metaData, MemoryStream mstream) in /mint/run/core/minio-dotnet/FunctionalTest.cs:line 923
   --- End of inner exception stack trace ---
   at System.Threading.Tasks.Task.Wait(Int32 millisecondsTimeout, CancellationToken cancellationToken)
   at System.Threading.Tasks.Task.Wait()
   at Minio.Functional.Tests.FunctionalTest.RemoveObjects_Test2(MinioClient minio) in /mint/run/core/minio-dotnet/FunctionalTest.cs:line 2134
   --- End of inner exception stack trace ---
   at System.Threading.Tasks.Task.Wait(Int32 millisecondsTimeout, CancellationToken cancellationToken)
   at System.Threading.Tasks.Task.Wait()
   at Minio.Functional.Tests.FunctionalTest.Main(String[] args) in /mint/run/core/minio-dotnet/FunctionalTest.cs:line 221

Executed 6 out of 13 tests successfully.
```

**Comment by krishnasrinivas** [comment]
@harshavardhana refactored a bit to make the code more intuitive

**Review by harshavardhana — APPROVED** [review]

**Review by kannappanr — APPROVED** [review]
LGTM

**Review by vadmeste — APPROVED** [review]
LGTM

**Pull request merged**